crates/sui-core/src/consensus_handler.rs

```diff
@@ -230,6 +239,37 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
     }
 }
 
+struct AsyncTransactionScheduler {
+    sender: tokio::sync::mpsc::Sender<Vec<VerifiedExecutableTransaction>>,
+}
+
+impl AsyncTransactionScheduler {
+    pub fn start(
+        transaction_manager: Arc<TransactionManager>,
+        epoch_store: Arc<AuthorityPerEpochStore>,
+    ) -> Self {
+        let (sender, recv) = tokio::sync::mpsc::channel(16);
+        spawn_monitored_task!(Self::run(recv, transaction_manager, epoch_store));
+        Self { sender }
+    }
+
+    pub async fn schedule(&self, transactions: Vec<VerifiedExecutableTransaction>) {
+        self.sender.send(transactions).await.ok();
```

Comment:
can we make the ok() an expect()?

Reply:
I think we do want to know if consensus output gets dropped before enqueue to transaction manager. If this particular case will never error, we may as well expect() here.

Reply:
hmm, yeah, this will be difficult to handle in the tests. i would settle for an error log here